rllib/agents/trainer.py

```diff
@@ -779,19 +791,41 @@ def env_creator_from_classpath(env_context):
             # Assert that user has not unset "in_evaluation".
             assert "in_evaluation" not in extra_config or \
                 extra_config["in_evaluation"] is True
+            # Merge user-provided eval config with the base config.
             evaluation_config = merge_dicts(self.config, extra_config)
             # Validate evaluation config.
             self.validate_config(evaluation_config)
-            # Switch on complete_episode rollouts (evaluations are
-            # always done on n complete episodes) and set the
-            # `in_evaluation` flag. Also, make sure our rollout fragments
-            # are short so we don't have more than one episode in one rollout.
-            evaluation_config.update({
-                "batch_mode": "complete_episodes",
-                "rollout_fragment_length": 1,
-                "in_evaluation": True,
-            })
+
+            # Set the `in_evaluation` flag.
+            evaluation_config["in_evaluation"] = True
+
+            # Evaluation duration unit: episodes.
+            # Switch on `complete_episode` rollouts. Also, make sure
+            # rollout fragments are short so we never have more than one
+            # episode in one rollout.
+            if evaluation_config["evaluation_duration_unit"] == "episodes":
+                evaluation_config.update({
+                    "batch_mode": "complete_episodes",
+                    "rollout_fragment_length": 1,
+                })
+            # Evaluation duration unit: timesteps.
+            # Set `batch_mode=truncate_episodes` and set
+            # `rollout_fragment_length` such that desired steps are divided
+            # equally amongst workers or - in auto duration mode - set it
+            # to a reasonable small number (10).
```

Comment:
actually any chance you can expand this comment a little bit, so we understand why a reasonable small number works for auto mode ... thanks :)

Reply:
done